common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -547,48 +547,26 @@ private boolean runShutdownHooks() {
             throw new NullPointerException("unit");
         }
 
-        if (isShuttingDown()) {
-            return terminationFuture();
-        }
-
-        boolean inEventLoop = inEventLoop();
-        boolean wakeup;
         int oldState;
         for (;;) {
-            if (isShuttingDown()) {
-                return terminationFuture();
-            }
-            int newState;
-            wakeup = true;
             oldState = STATE_UPDATER.get(this);
-            if (inEventLoop) {
-                newState = ST_SHUTTING_DOWN;
-            } else {
-                switch (oldState) {
-                    case ST_NOT_STARTED:
-                    case ST_STARTED:
-                        newState = ST_SHUTTING_DOWN;
-                        break;
-                    default:
-                        newState = oldState;
```

Comment:
this is a change in behavior. the new code will always try to change the state but the old code, but the old code only changed the state in the event loop. @normanmaurer - WDYT (here is the original change [1])

[1] https://github.com/netty/netty/commit/16a85e6cca46cfdcfe07c9e76a3b935c72c5ec1d#diff-cde1b8aa1fe0c3a083412f79421bafcfR501